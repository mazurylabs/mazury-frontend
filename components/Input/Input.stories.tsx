import { ComponentMeta, ComponentStory } from '@storybook/react';
import * as React from 'react';
import { useState } from 'react';

import { Input } from './Input';

export default {
  title: 'Components/Input',
  component: Input,
  argTypes: {
    id: { control: { type: 'text' } },
    placeholder: { control: { type: 'text' } },
    label: { control: { type: 'text' } },
    type: { control: { type: 'text' } },
    disabled: { control: { type: 'boolean' } },
  },
} as ComponentMeta<typeof Input>;

const Template: ComponentStory<typeof Input> = (args) => {
  const [value, setValue] = useState('');

  return <Input {...args} value={value} onChange={setValue} />;
};

export const Default = Template.bind({});
Default.args = {
  id: 'input-1',
  placeholder: 'Placeholder',
  label: 'Label',
  type: 'text',
};

export const Disabled = Template.bind({});
Disabled.args = {
  id: 'input-1',
  type: 'text',
  disabled: true,
};

export const Error = Template.bind({});
Error.args = {
  id: 'input-1',
  type: 'text',
  error: true,
};
