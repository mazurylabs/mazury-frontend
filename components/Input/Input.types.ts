import { HTMLInputTypeAttribute, ReactNode } from 'react';

export interface InputProps {
  label?: ReactNode;
  value?: string;
  onChange?: (value: string) => void;
  placeholder?: string;
  id?: string;
  type?: HTMLInputTypeAttribute;
  className?: string;
  outerClassName?: string;
  disabled?: boolean;
  error?: boolean;
}
