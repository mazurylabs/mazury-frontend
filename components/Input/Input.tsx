import { FC } from 'react';
import { InputProps } from './Input.types';

export const Input: FC<InputProps> = ({
  value,
  onChange,
  placeholder,
  label,
  id,
  type = 'text',
  className,
  /*
  The className that will be applied to the outer div containing the label and the input
  */
  outerClassName,
  disabled = false,
  error = false,
}) => {
  return (
    <div className={`flex flex-col ${outerClassName}`}>
      {label && (
        <label className="text-sm font-medium text-indigoGray-40" htmlFor={id}>
          {label}
        </label>
      )}
      <input
        id={id}
        type={type}
        value={value}
        onChange={(e) => {
          if (onChange) {
            onChange(e.target.value);
          }
        }}
        placeholder={placeholder}
        className={`block rounded-lg border-[1.5px] border-indigoGray-30 px-4 py-3 text-base font-medium text-indigoGray-90 outline-none placeholder:text-indigoGray-50 hover:border-indigoGray-40 focus:border-indigoGray-50 ${
          disabled && 'cursor-not-allowed border-indigoGray-30 bg-indigoGray-10'
        } ${error && 'border-red-500'} ${className}`}
        disabled={disabled}
      />
    </div>
  );
};
