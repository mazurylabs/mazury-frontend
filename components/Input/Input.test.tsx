import { render } from '@testing-library/react';
import { Input } from './Input';

describe('Input', () => {
  test('input with label and placeholder', () => {
    const { getByLabelText, getByPlaceholderText } = render(
      <Input id="input-1" placeholder="Placeholder" label="Label" />
    );
    expect(getByLabelText('Label')).toBeInTheDocument();
    expect(getByPlaceholderText('Placeholder')).toBeInTheDocument();
    expect(getByLabelText('Label')).toMatchSnapshot();
  });

  test('disabled', () => {
    const { getByPlaceholderText } = render(
      <Input id="input-1" placeholder="Placeholder" label="Label" disabled />
    );
    const input = getByPlaceholderText('Placeholder');
    expect(input).toBeInTheDocument();
    expect(input).toBeDisabled();
    expect(input).toMatchSnapshot();
  });

  test('error', () => {
    const { getByPlaceholderText } = render(
      <Input id="input-1" placeholder="Placeholder" label="Label" error />
    );
    const input = getByPlaceholderText('Placeholder');
    expect(input).toBeInTheDocument();
    expect(input).toHaveClass('border-red-500');
    expect(input).toMatchSnapshot();
  });
});
