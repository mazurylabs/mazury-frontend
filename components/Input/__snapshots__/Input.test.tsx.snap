// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input disabled 1`] = `
<input
  class="block rounded-lg border-[1.5px] border-indigoGray-30 px-4 py-3 text-base font-medium text-indigoGray-90 outline-none placeholder:text-indigoGray-50 hover:border-indigoGray-40 focus:border-indigoGray-50 cursor-not-allowed border-indigoGray-30 bg-indigoGray-10 false undefined"
  disabled=""
  id="input-1"
  placeholder="Placeholder"
  type="text"
  value=""
/>
`;

exports[`Input error 1`] = `
<input
  class="block rounded-lg border-[1.5px] border-indigoGray-30 px-4 py-3 text-base font-medium text-indigoGray-90 outline-none placeholder:text-indigoGray-50 hover:border-indigoGray-40 focus:border-indigoGray-50 false border-red-500 undefined"
  id="input-1"
  placeholder="Placeholder"
  type="text"
  value=""
/>
`;

exports[`Input input with label and placeholder 1`] = `
<input
  class="block rounded-lg border-[1.5px] border-indigoGray-30 px-4 py-3 text-base font-medium text-indigoGray-90 outline-none placeholder:text-indigoGray-50 hover:border-indigoGray-40 focus:border-indigoGray-50 false false undefined"
  id="input-1"
  placeholder="Placeholder"
  type="text"
  value=""
/>
`;
